python/lsst/pipe/tasks/calibrate.py

```diff
@@ -478,7 +480,7 @@ def __init__(self, butler=None, astromRefObjLoader=None,
 
     def runQuantum(self, butlerQC, inputRefs, outputRefs):
         inputs = butlerQC.get(inputRefs)
-        inputs['exposureIdInfo'] = ExposureIdInfo.fromDataId(butlerQC.quantum.dataId, "visit_detector")
+        inputs['idGenerator'] = self.config.idGenerator.apply(butlerQC.quantum.dataId)
```

Comment:
I agree with you on both points, especially in terms of recommendations for new code rather than something to try to fix in existing tasks.  But I definitely don't want to attempt that on this ticket.

Reply:
Would you mind filing a ticket about this? Something to encompass both implementing the change to "Our convention is to call `run` from `runQuantum` with `**inputs`, as extracted from the quantum graph, plus separate named kwargs." in existing code, and also documenting it somewhere (the `PipelineTask` docs?).

Reply:
I'm sure I actually agree it's worth making a convention out of.  I agree I generally prefer what you've said, but it's not like the alternative is in any way problematic (even in terms of readability) and I'm wary of over-prescription in conventions.